bootstrap/httpserver/status.go

```diff
@@ -0,0 +1,80 @@
+package httpserver
+
+import (
+	"encoding/json"
+	"github.com/orbs-network/orbs-network-go/instrumentation/metric"
+	"github.com/orbs-network/scribe/log"
+	"net/http"
+	"time"
+)
+
+type StatusResponse struct {
+	Timestamp time.Time
+	Status    string
+	Error     string
+	Payload   map[string]metric.ExportedMetric
+}
+
+func (s *HttpServer) getStatus(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	status := StatusResponse{
+		Timestamp: time.Now(),
+		Status:    s.getStatusWarningMessage(),
+		Payload:   s.metricRegistry.ExportAll(),
+	}
+
+	data, _ := json.MarshalIndent(status, "", "  ")
+
+	_, err := w.Write(data)
+	if err != nil {
+		s.logger.Info("error writing index.json response", log.Error(err))
+
+	}
+
+}
+
+func (s *HttpServer) getStatusWarningMessage() string {
+	maxTimeSinceLastBlock := s.config.TransactionPoolTimeBetweenEmptyBlocks().Nanoseconds() * 10
+	if maxTimeSinceLastBlock < 600000000 { // ten minutes
+		maxTimeSinceLastBlock = 600000000
+	}
+	if s.getGaugeValueFromMetrics("ConsensusAlgo.LeanHelix.LastCommitted.TimeNano")+maxTimeSinceLastBlock <
+		time.Now().UnixNano() {
+		return "Last Successful Committed Block was too long ago"
+	}
+
+	if len(s.config.ManagementFilePath()) != 0 && s.config.ManagementPollingInterval() > 0 {
+		maxIntervalSinceLastSuccessfulManagementUpdate := int64(s.config.ManagementPollingInterval().Seconds()) * 20
+		if s.metricRegistry.Get("Management.Data.LastSuccessfulUpdateTime").Export().LogRow()[0].Value().(int64)+maxIntervalSinceLastSuccessfulManagementUpdate <
```

Comment:
in `getGaugeValueFromMetrics ` you protect against metric names that are not yet defined. could this line panic too? consider moving panic safety to this method?...

Reply:
I am using `getGaugeValueFromMetrics` now instead.